core/src/main/java/org/pentaho/di/core/row/value/ValueMetaBase.java

```diff
@@ -0,0 +1,5534 @@
+        //
+        String[] emptyPaddedStrings = Const.getEmptyPaddedStrings();
+        if ( length < emptyPaddedStrings.length ) {
+          return emptyPaddedStrings[length];
+        } else {
+          return Const.rightPad( "", length );
+        }
+      }
+    }
+
+    try {
+      return getDecimalFormat( false ).format( integer );
+    } catch ( Exception e ) {
+      throw new KettleValueException( toString() + " : couldn't convert Long to String ", e );
+    }
+  }
+
+  protected synchronized String convertIntegerToCompatibleString( Long integer ) throws KettleValueException {
+    if ( integer == null ) {
+      return null;
+    }
+    return Long.toString( integer );
+  }
+
+  protected synchronized Long convertStringToInteger( String string ) throws KettleValueException {
+    string = Const.trimToType( string, getTrimType() ); // see if trimming needs
+    // to be performed before
+    // conversion
+
+    if ( Utils.isEmpty( string ) ) {
+      return null;
+    }
+
+    try {
+      Number number;
+      if ( lenientStringToNumber ) {
+        number = new Long( getDecimalFormat( false ).parse( string ).longValue() );
+      } else {
+        ParsePosition parsePosition = new ParsePosition( 0 );
+        number = getDecimalFormat( false ).parse( string, parsePosition );
+
+        if ( parsePosition.getIndex() < string.length() ) {
+          throw new KettleValueException( toString()
+              + " : couldn't convert String to number : non-numeric character found at position "
+              + ( parsePosition.getIndex() + 1 ) + " for value [" + string + "]" );
+        }
+
+      }
+      return new Long( number.longValue() );
+    } catch ( Exception e ) {
+      throw new KettleValueException( toString() + " : couldn't convert String to Integer", e );
+    }
+  }
+
+  protected synchronized String convertBigNumberToString( BigDecimal number ) throws KettleValueException {
+    if ( number == null ) {
+      return null;
+    }
+
+    try {
+      return getDecimalFormat( bigNumberFormatting ).format( number );
+    } catch ( Exception e ) {
+      throw new KettleValueException( toString() + " : couldn't convert BigNumber to String ", e );
+    }
+  }
+
+  protected synchronized BigDecimal convertStringToBigNumber( String string ) throws KettleValueException {
+    string = Const.trimToType( string, getTrimType() ); // see if trimming needs
+    // to be performed before
+    // conversion
+
+    if ( Utils.isEmpty( string ) ) {
+      return null;
+    }
+
+    try {
+      DecimalFormat format = getDecimalFormat( bigNumberFormatting );
+      Number number;
+      if ( lenientStringToNumber ) {
+        number = format.parse( string );
+      } else {
+        ParsePosition parsePosition = new ParsePosition( 0 );
+        number = format.parse( string, parsePosition );
+
+        if ( parsePosition.getIndex() < string.length() ) {
+          throw new KettleValueException( toString()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Exception [KettleValueException] is explicitly thrown on this line but is caught in the outer catch clause with generic exception on line [1406].  Consider adding a catch clause of this specific exception in the outer try statement.